File: Capítulo_14/Introdução à programação funcional/ES2015+ e a programação funcional/findMinArray.js
// Programação imperativa:
var findMinArray = function(array) {
    var minValue = array[0];
    for (var i = 1; i < array.length; i++) {
        if (minValue > array[i]) {
            minValue = array[i];
        }
    }
    return minValue;
};
console.log(findMinArray([8, 6, 4, 5, 9])); // exibe 4

// Programação funcional:
const min_ = function(array) {
    return Math.min(...array);
};
console.log(min_([8, 6, 4, 5, 9])); // exibe 4

// Funções de seta (arrow functions):
const min = arr => Math.min(...arr);
console.log(min([8, 6, 4, 5, 9])); // exibe 4

File: Capítulo_14/Introdução à programação funcional/ES2015+ e a programação funcional/Código em Python/findMinArray.py
# Programação imperativa:
def find_min_array(array):
    min_value = array[0]
    for i in range(1, len(array)):
        if min_value > array[i]:
            min_value = array[i]

    return min_value

print(find_min_array([8, 6, 4, 5, 9])) # exibe 4

# Programação funcional:
def min_(array):
    return min(*array)

print(min_([8, 6, 4, 5, 9])) # exibe 4

# Funções lambda:
min_value = lambda arr: min(arr)
print(min_value([8, 6, 4, 5, 9])) # exibe 4